bigc/resources/customers.py

```diff
@@ -81,6 +85,40 @@ def update_form_field(self, customer_id: int, field_name: str, value: Any) -> di
         }]
         return self._api.v3.put('/customers/form-field-values', json=payload)[0]
 
-    def delete(self, customer_id: int) -> None:
-        """Delete a specific customer by its ID"""
-        self._api.v3.delete(f'/customers?id:in={customer_id}')
+    def all_addresses(self, customer_id: int) -> Iterator[dict]:
+        """Get all addresses, filtering for a single customer"""
+        return self._api.v3.get_many(f'/customers/addresses?customer_id:in={customer_id}')
+
+    def get_address(self, customer_id: int, address_id: int) -> dict:
+        """Get one address by its ID, filtering for a single customer"""
+        return self._api.v3.get(f'/customers/addresses?customer_id:in={customer_id}&id:in={address_id}')
+
+    def create_address(self, address_data: dict) -> dict:
+        """Create an address"""
+        valid_address_params = ['customer_id', 'first_name', 'last_name', 'company', 'phone', 'address_type',
+                                'address1', 'address2', 'city', 'country_code', 'state_or_province', 'postal_code']
+
+        validated_address_data = [{
+            address_key: address_value
+            for address_key, address_value in address_data.items()
+            if address_key in valid_address_params
+        }]
+
+        return self._api.v3.post('/customers/addresses', json=validated_address_data)
```

Comment:
I don't really know what the point of this validation is. If the user provides a key that BigCommerce doesn't expect, then we should just let BigCommerce handle that. We don't want to put our own validation in the middle.

Here's how I'd expect this to work, based on how we've done things in this library so far:

```suggestion
    def create_address(self, customer_id: int, **kwargs) -> dict:
        """Add an address to the customer's address book"""
        payload = {
            'customer_id': customer_id,
            **kwargs,
        }

        return self._api.v3.post('/customers/addresses', json=payload)
```

Reply:
We're not "throwing our own" error, moreso just cleaning the data that gets passed to Bigcommerce but I see your point